### PR TITLE
close #1457 add tabsusecases for removing all/privateonly/nonprivate sessions

### LINKED ISSUE #1457: Add new TabsUseCases for closing all sessions
We need a use case that lets you close:
- all
- regular tabs only
- private tabs only

### PR DESCRIPTION
**Pull request by chikka**
add tabsusecases for removing all/privateonly/nonprivate sessions

### CODE REVIEW

File: components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt [hunk 1]
```diff
@@ -93,8 +93,48 @@ class TabsUseCases(
...
+            sessionManager.removeAll()
+
+        }
+    }
+
+    class RemovePrivateTabsUseCase internal constructor(
+            private val sessionManager: SessionManager
+    ) {
+
+        fun invoke() {
```
Comment (jonalmeida) [inline components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt]: You could consider pass a boolean argument to the invoke function to differ between removing regular/private tabs.

File: components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt [hunk 2]
```diff
@@ -93,8 +93,48 @@ class TabsUseCases(
         }
     }
 
+    class RemoveAllTabsUseCase internal constructor(
```
Comment (jonalmeida) [inline components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt]: If you could write a few basic tests for these use cases as well, that would be great too. (: 

We have some handy ones in [`TabsUseCasesTest.kt`](https://github.com/mozilla-mobile/android-components/blob/master/components/feature/tabs/src/test/java/mozilla/components/feature/tabs/TabsUseCasesTest.kt) for reference.

File: components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt (hunk 1 above)
Comment (chikka) [reply components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt]: ok.will add

File: components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt (hunk 2 above)
Comment (chikka) [reply components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt]: sure

File: components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt (hunk 1 above)
Comment (chikka) [reply components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt]: sure

**Pull request closed without merging**